### PR TITLE
Added the running median function

### PR DESCRIPTION
**Pull request by tristanmech**
Function that computes the running median for n-d arrays. Need to add a test

### CODE REVIEW

File: spectrapepper/functions.py [hunk 1]
```diff
@@ -2806,3 +2806,45 @@ def asymmetry(data, peak, axis, res= 0.001, s=5, limit=10):
...
+    data_new2 = np.c_[data[:,:pad_length],data_new]
+    
+    # Get the windows
+    window = np.lib.stride_tricks.sliding_window_view(data_new2,window_size)
+    
+    # Create new spectra from the median of each window
+    new_spectra = []
+    for i in window[0]:
+        new_spectra.append(np.median(i))
+    new_spectra = np.asarray(new_spectra)
```
Comment (ksunden) [inline spectrapepper/functions.py:2846]:
```suggestion
    new_spectra = np.median(window, axis=-1)
```

No need to append to a list then reconvert back to ndarray, numpy can do that much for you:


see the example (for mean) at the bottom of the sliding_window_view docs:
https://numpy.org/doc/stable/reference/generated/numpy.lib.stride_tricks.sliding_window_view.html?highlight=moving%20average

File: spectrapepper/functions.py [hunk 2]
```diff
@@ -2806,3 +2806,45 @@ def asymmetry(data, peak, axis, res= 0.001, s=5, limit=10):
...
+    :type plot: boolean
+    :param plot: If you want to plot the new spectra change to True
+
+    :returns: Array containing the computed 1D spectra.
+    :rtype: numpy array[float]
+    """
+    # Check dimensions
+    if np.shape(window_size)[0] != 2:
+        raise ValueError("The window must be 2D")
+    if (window_size[1] % 2) == 0:
```
Comment (ksunden) [inline spectrapepper/functions.py:2832]: seems arbitrary restriction, median is perfectly well defined for even number of elements

Comment (tristanmech) [reply spectrapepper/functions.py:2832]: The idea behind that restriction was to keep the spectra from shifting left or right after the processing

File: spectrapepper/functions.py (hunk 1 above)
Comment (tristanmech) [reply spectrapepper/functions.py:2846]: Thanks for pointing that out, I did not notice it when I was reading the docs

**Comment by enricgrau** [comment]
@tristanmech Checks are failing for Ubunto 3.6. Please review this so it does pass. Also, please include a better discription of the function, that is, better narrate the mathematical procedure. 1-2 sentences should be more than enugh for this. Efharisto! :)

**Comment by ksunden** [comment]
`sliding_window_view` is numpy 1.20+

If you wish to follow [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html) numpy 1.18+ is recommended currently.

**Comment by enricgrau** [comment]
@tristanmech Please, check the Ubunto 3.6 build fail!

**Comment by tristanmech** [comment]
Will do
‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by tristanmech** [comment]
The problem seems to be that numpy.lib.stride_tricks.sliding_window_view was introduced on numpy 1.20 and the Ubuntu 3.6 build uses an older version (numpy 1.19.5). From what I can understand looking at it, this is a pre-baked Ubuntu VM, so can we ask for it to install a greater than 1.2 version of numpy or it simply does not support it?

**Comment by tristanmech** [comment]
They are fine except for a couple of points:
1. The mkvirtualenv command does not work on Windows just by installing virtualenv and virtualenvwrapper, you need to install virtualenvwrapper-win and then it is fine [ref](https://docs.djangoproject.com/en/2.2/howto/windows/)
2. The tox command does not work without a config file which by default is not provided by the repo. The error is message is: "ERROR: tox config file (either pyproject.toml, tox.ini, setup.cfg) not found"
3. Not really a problem but for verbosity maybe add a "cd tests" at step 6.

Question: Why is virtualenv used instead of conda?

**Comment by enricgrau** [comment]
@tristanmech Thank you. I've created a [new issue](https://github.com/spectrapepper/spectrapepper/issues/10) to discuss this as is not related to this pull request (my bad). To continue with the original thread, have you been able to work on the ubuntu 3.6 check fail?

**Comment by tristanmech** [comment]
I think adding numpy >= 1.20 in the requirements should fix the problem if those VMs are configurable in this way.

**Comment by enricgrau** [comment]
> I think adding numpy >= 1.20 in the requirements should fix the problem if those VMs are configurable in this way.

Try to add it yourself and retry the tests ;)

**Pull request merged**